compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1351,15 +1351,23 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                             *in_trait,
                             itctx,
                         ),
-                    ImplTraitContext::TypeAliasesOpaqueTy => self.lower_opaque_impl_trait(
-                        span,
-                        hir::OpaqueTyOrigin::TyAlias,
-                        def_node_id,
-                        bounds,
-                        false,
-                        &ImplTraitContext::TypeAliasesOpaqueTy,
-                    ),
+                    ImplTraitContext::TypeAliasesOpaqueTy(parent_def_id) => self
+                        .lower_opaque_impl_trait(
+                            span,
+                            hir::OpaqueTyOrigin::TyAlias(*parent_def_id),
+                            def_node_id,
+                            bounds,
+                            false,
+                            itctx,
+                        ),
                     ImplTraitContext::Universal => {
+                        if self.opt_local_def_id(def_node_id).is_none() {
```

Comment:
I what case would this ever be `Some`?

Reply:
Same as https://github.com/rust-lang/rust/pull/102483#discussion_r984017996